### PR TITLE
chore(CPlusPlus): add reverse number algorithm

### PR DESCRIPTION
**Pull request by Ashad001**
**PR Checklist:**
- [x] My submission is formatted according to the guidelines in the [contributing guide](https://github.com/MakeContributions/DSA#contribution-guidelines)
- [x] My addition is on refer on the language README.md file
- [x] My addition does not have a spelling problem
- [x] My submission has a proper and user-friendly description of the algorithm
- [x] My submission has the time complexity of the algorithm
- [x] My submission has sample input-output of the program (NOT FOR PYTHON)

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [x] New algorithm
- [ ] Optimization in previous algorithms
- [ ] Code style update
- [ ] Refactor
- [ ] Documentation
- [ ] Other, please describe:

**Briefly describe the changes in this PR**

**Other information:**

### CODE REVIEW

**Comment by taimoon** [comment]
IMHO, the reliance on static variables can cause confusing bugs to anyone who does not fully grasp the concept of lifetime and scoping (including me). Indeed, the `Rev` does not give expected results when I try to call it multiple times on my platform.

Furthermore, the computation of log is expensive.

Below is another way to implement recursive reverse function without using `log`. Of course, it can be easily transformed to iterative version. 

```
int RevHelper(int n, int radix, int curr){
	if(n < 1)
		return curr;
	else
		return RevHelper(n/radix, radix, curr*radix + n%radix);
}
int Rev(int n){
	return RevHelper(n, 10, 0);
}
```

**Comment by Ashad001** [comment]
> IMHO, the reliance on static variables can cause confusing bugs to anyone who does not fully grasp the concept of lifetime and scoping (including me). Indeed, the `Rev` does not give expected results when I try to call it multiple times in the `main` function on my platform.
> 
> Furthermore, the computation of log is expensive.
> 
> Below is another way to implement recursive reverse function without using `log`. Of course, it can be easily transformed to iterative version.
> 
> ```
> int RevHelper(int n, int radix, int curr){
> 	if(n < 1)
> 		return curr;
> 	else
> 		return RevHelper(n/radix, radix, curr*radix + n%radix);
> }
> int Rev(int n){
> 	return RevHelper(n, 10, 0);
> }
> ```
> 
> Using lambda expression,
> 
> ```
> #include<functional>
> int Rev(int n){
> 	function<int(int,int, int)> helper = [&](int n, int radix, int curr){
> 		if(n < 1)
> 			return curr;
> 		else
> 			return helper(n/radix, radix, curr*radix + n%radix);
> 	};
> 	return helper(n, 10, 0);
> }
> ```

Didn't know about the drawbacks of using static variables. Thanks for sharing :)

**Comment by Ashad001** [comment]
Can someone explain me this fail I am getting?
image

File: algorithms/CPlusPlus/Recursion/reverse-a-number.cpp [hunk 1]
```diff
@@ -0,0 +1,34 @@
+/*
+Description: Program to reverse integer using recursion
+Time complexity: O(n) where n is the number of digits in the integer
+*/
+
+#include <iostream>
+#include <math.h>
```
Comment (Arsenic-ATG) [inline algorithms/CPlusPlus/Recursion/reverse-a-number.cpp:7]:
```suggestion
#include <cmath>
#include <iostream>
```

This one is causing the linting checks to fail:
prefer cpp style header and put c headers before cpp headers

**Review by Arsenic-ATG — CHANGES_REQUESTED** [review]
Apart from the review mentioned in review comments, the PR also need to update the proper README.md with the entry of the program in it. 


For your particular program it would be [algorithms/CPlusPlus/README.md](https://github.com/Ashad001/DSA/blob/add-file/algorithms/CPlusPlus/README.md) under the **Recursion** section.

**Comment by Arsenic-ATG** [comment]
Now only update the readme.md and the PR should be good to go

**Comment by Ashad001** [comment]
Done! I think that's it. Thanks for helping me out!

**Review by Arsenic-ATG — APPROVED** [review]

**Comment by ming-tsai** [comment]
Hi @Arsenic-ATG,
The issue of `cpplint` pipeline is fixed on this pull request #468?

**Comment by Arsenic-ATG** [comment]
> Hi @Arsenic-ATG,
> The issue of `cpplint` pipeline is fixed on this pull request #468?

The check was failing due to the "fibonacci number" project (added in #794 ), due to some reasons this was not detected there.

My last commit with this PR should fix that.

**Review by Arsenic-ATG — APPROVED** [review]

**Comment by ming-tsai** [comment]
Thanks, @Arsenic-ATG nice work!!!

**Pull request merged**